### ISSUE TITLE
NaN output on calculateLoss()

### ISSUE THREAD

**Issue by joobei** [jordenhill/Birdbrain#9]
Jorden I tried to train your rnn with a very simple method:

Target: 1 for random positive gaussian numbers
Target: 0 for random negative gaussian numbers

Loss seems to become not-a-number after a few training runs. If you're interested in getting to the bottom of this then please run the following playground alongside birdbrain:

var rnn = RecurrentNeuralNetwork(inputDim: 2,hiddenDim: 1,useMetal: false,activationFunction:"tangent")

```
for x in 0...1000 {
    //generate a random positive float
    let pos = abs(rand_gauss())
    //generate a random negative float
    let neg = abs(rand_gauss())*(-1)
    //train the RNN
    rnn.backprop([[pos]], target: [Array(count: 1, repeatedValue: 1)], learningRate: 0.001)
    rnn.backprop([[neg]], target: [Array(count: 1, repeatedValue: 0)], learningRate: 0.001)
    var loss = rnn.calculateLoss([[abs(rand_gauss())]], target: [[1]], numExamples: 1)
    Swift.print(loss)
}

//attempt at basic prediction
let pos = abs(rand_gauss())
let neg = abs(rand_gauss())*(-1)

Swift.print(rnn.predict([[pos]]))
Swift.print(rnn.predict([[neg]]))

```
